src/main/java/de/mediathekview/mserver/crawler/dw/parser/DwFilmDetailDeserializer.java

```diff
@@ -0,0 +1,226 @@
+      return Optional.empty();
+    }
+    if (!type.get().equalsIgnoreCase("video")) {
+    	//LOG.error("element is not a video {} but {}", videoId.orElse(title.orElse(topic.orElse(""))), type.orElse(""));
+        return Optional.empty();    	
+    }
+    
+    if (!jsonObject.has(ELEMENT_MAINCONTENT)) {
+      LOG.error("Could not find maincontent for video {}", videoId.get());
+      return Optional.empty();
+    }
+
+    final JsonObject jsonObjectMainContent = jsonObject.get(ELEMENT_MAINCONTENT).getAsJsonObject();
+    final Optional<String> thisPageUrl = JsonUtils.getAttributeAsString(jsonObjectMainContent, ELEMENT_MAINCONTENT_LINK);
+    
+    if (!jsonObjectMainContent.has(ELEMENT_MAINCONTENT_SOURCES)) {
+      LOG.error("Could not find sources for video {} url {}", videoId.get(), thisPageUrl.orElse(""));
+      return Optional.empty();
+    }
+    if (thisPageUrl.isEmpty()) {
+      LOG.error("Could not find thisPageUrl for video {}", videoId.get());
+      return Optional.empty();
+    }
+    
+    final Film film =
+        new Film(UUID.randomUUID(), sender, title.get(), topic.get(), getAiredDate(thisPageUrl.get(),jsonObject), getDuration(thisPageUrl.get(),jsonObjectMainContent));
+    //
+    final Optional<String> description = JsonUtils.getAttributeAsString(jsonObject, ELEMENT_TEASER);
+    description.ifPresent(film::setBeschreibung);
+    //
+    getWebsite(thisPageUrl.get(), jsonObject).ifPresent(film::setWebsite);
+    //
+    final JsonArray jsonObjectMainContentSources = jsonObjectMainContent.get(ELEMENT_MAINCONTENT_SOURCES).getAsJsonArray();
+    getVideos(title.get(),jsonObjectMainContentSources).ifPresent(film::addAllUrls);
+    //
+    return Optional.of(film);
+  }
+  
+  private Optional<URL> getWebsite(final String videoid, final JsonObject jsonObject) {
+    Optional<URL> websiteUrl = Optional.empty();
+    final Optional<String> websiteString = JsonUtils.getAttributeAsString(jsonObject, ELEMENT_LINK);
+    if (websiteString.isPresent()) {
+      try {
+        websiteUrl = Optional.of(new URL(websiteString.get()));
+      } catch (Exception e) {
+        LOG.error("Error getWebsite for video {} on value '{}'", videoid, websiteString.get());
+      }
+    } else {
+      LOG.error("no error getWebsite found for video {}", videoid);
+    }
+    return websiteUrl;
+  }
+  
+  private LocalDateTime getAiredDate(final String videoid, final JsonObject jsonObject) {
+    final DateTimeFormatter dateFormatter =
+        DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSX");
+    LocalDateTime airedDatetime = LocalDateTime.now();
+    final Optional<String> displayDate = JsonUtils.getAttributeAsString(jsonObject, ELEMENT_DATETIME);
+    if (displayDate.isPresent()) {
+      try {
+        airedDatetime = LocalDateTime.parse(displayDate.get(), dateFormatter);
```

Comment:
die Uhrzeit ist wenig aussagekräftig, scheint in der Regel die Zeit, zu der es online gestellt wurde zu sein. Willst du das so lassen oder immer 00:00:00 als Uhrzeit nehmen?

Reply:
Ja würde ich so lassen - ich gebe mir hier soviel mühe ein Datum (plus Zeit) zu bekommen :-)